src/main/java/org/trigon/hardware/rev/spark/io/SimulationSparkIO.java

```diff
@@ -1,60 +1,74 @@
 package org.trigon.hardware.rev.spark.io;
 
-import com.revrobotics.*;
+import com.revrobotics.sim.SparkAbsoluteEncoderSim;
+import com.revrobotics.spark.SparkBase;
+import com.revrobotics.spark.SparkClosedLoopController;
+import com.revrobotics.spark.SparkMax;
+import com.revrobotics.spark.SparkSim;
+import com.revrobotics.spark.config.SparkBaseConfig;
+import com.revrobotics.spark.config.SparkMaxConfig;
 import edu.wpi.first.math.system.plant.DCMotor;
+import org.trigon.hardware.RobotHardwareStats;
 import org.trigon.hardware.rev.spark.SparkIO;
 import org.trigon.hardware.rev.sparkecnoder.SparkEncoder;
+import org.trigon.hardware.simulation.MotorPhysicsSimulation;
+import org.trigon.utilities.Conversions;
 
 public class SimulationSparkIO extends SparkIO {
-    private final CANSparkMax motor;
-    private final SparkPIDController pidController;
-    private final SparkEncoder encoder;
+    private final SparkMax motor;
+    private final SparkSim motorSimulation;
+    private final SparkClosedLoopController pidController;
+    private final SparkAbsoluteEncoderSim absoluteEncoderSimulation;
+    private SparkEncoder encoder = null;
+    private MotorPhysicsSimulation physicsSimulation = null;
+    private boolean isUsingAbsoluteEncoder = false;
 
-    public SimulationSparkIO(int id, DCMotor gearbox) {
-        motor = new CANSparkMax(id, CANSparkMax.MotorType.kBrushless);
-        REVPhysicsSim.getInstance().addSparkMax(motor, gearbox);
-        pidController = motor.getPIDController();
-        encoder = SparkEncoder.createRelativeEncoder(motor);
+    public SimulationSparkIO(int id) {
+        motor = new SparkMax(id, SparkMax.MotorType.kBrushless);
+        encoder = SparkEncoder.createAbsoluteEncoder(motor);
+        pidController = motor.getClosedLoopController();
+        motorSimulation = new SparkSim(motor, DCMotor.getBag(1)); /** DCMotor.getBag(1) is a placeholder, we don't actually care about this since we always do {@link com.revrobotics.sim.SparkMaxSim#setMotorCurrent} */
```

Comment:
I don't like this (for obvious reasons) but if you do need to do this `/**` -> `/*`

Reply:
No. If I do it like that the link wont work.